### ISSUE TITLE
--add-host alias:ip should admit --add-host alias~ip:port

### ISSUE THREAD

**Issue by jackphel** [docker/docker#12576]
for development I want to do both
`docker run --add-host service:localhost service-consumer`
and 
`docker run --add-host service2:localhost service2-consumer`

but both services expect port 80. Apparently I can do something along the lines of 

```

sudo ifconfig lo0 10.0.0.1 alias
echo "rdr pass on lo0 inet proto tcp from any to 10.0.0.1 port 80 -> 127.0.0.1 port 3000" | sudo pfctl -ef -
docker run --add-host service:10.0.0.1 service-consumer
sudo ifconfig lo0 10.0.0.2 alias
echo "rdr pass on lo0 inet proto tcp from any to 10.0.0.2 port 80 -> 127.0.0.1 port 2000" | sudo pfctl -ef -
docker run --add-host service:10.0.0.2 service-consumer
```

but to be honest I can't even get this working and it's a pain if the external service turns out to use port 443 instead of 80. Who wants to figure out this hairy code?

TLDR
**It would be so great if docker would let me**
`docker run --add-host service~localhost:3000 service-consumer` 
`docker run --add-host service2~localhost:4000 service2-consumer`

**Comment by thaJeztah** [comment]
I don't think `--add-host` is really suitable for this, because all that does is adding extra entries to `/etc/hosts` for IP lookups.

You might be interested in the Network API that's currently being worked on and will be added as a preview/experimental feature soon (hopefully): https://github.com/docker/docker/pull/13441

I'm going to close this issue, because I don't expect this to be implemented this way, and alternatives are being worked on as part of https://github.com/docker/docker/pull/13441
